packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -462,10 +462,45 @@ class AndroidSdk {
     return versionString.split('_').first;
   }
 
-  /// Finds the java binary that is used for all operations across the tool.
+  /// A value that would be appropriate to use as JAVA_HOME.
   ///
-  /// First try Java bundled with Android Studio, then sniff JAVA_HOME, then fallback to PATH.
+  /// This method considers jdk in the following order:
+  /// * the JDK bundled with Android Studio, if one is found;
+  /// * the JAVA_HOME in the ambient environment, if set;
+  String? get javaHome {
+    return findJavaHome(
+      androidStudio: globals.androidStudio,
+      fileSystem: globals.fs,
+      operatingSystemUtils: globals.os,
+      platform: globals.platform,
+    );
+  }
+
+
+  static String? findJavaHome({
+    required AndroidStudio? androidStudio,
+    required FileSystem fileSystem,
+    required OperatingSystemUtils operatingSystemUtils,
+    required Platform platform,
+  }) {
+    if (androidStudio?.javaPath != null) {
+      globals.printTrace("Using Android Studio's java.");
+      return androidStudio!.javaPath!;
+    }
+
+    final String? javaHomeEnv = platform.environment[_javaHomeEnvironmentVariable];
+    if (javaHomeEnv != null) {
+      // Trust JAVA_HOME.
+      globals.printTrace('Using JAVA_HOME.');
```

Comment:
This log line and the comment no longer makes sense in the refactored method `findJavaHome`